### PR TITLE
Add possibility to run sidecars with vault

### PR DESCRIPTION
**Pull request by shahbazn**
Problem: This chart does not allow running sidecar containers with vault, that might me needed for some scenarios e.g. running prometheus exporter.

Solution: Add configurable sidecar containers to the chart

### CODE REVIEW

**Comment by ncabatoff** [comment]
You shouldn't need a prometheus exporter now that prometheus metrics can be scraped directly from Vault.

**Comment by ncabatoff** [comment]
> Could there be other cases of running auxiliary services with vault?

Sure, I wasn't suggesting the PR should be rejected, I was just letting you know in case you weren't aware that a prometheus exporter isn't strictly necessary.

**Comment by jasonodonnell** [comment]
Hi @shahbazn, you'll need to sign the CLA for us to consider this PR.  Thanks!

**Comment by shahbazn** [comment]
Hi @jasonodonnell I have added the unit tests now. Had to force push the initial commits to include my email address for CLA to work.

**Comment by shahbazn** [comment]
Hi @jasonodonnell the PR is ready to be reviewed.

**Comment by AnthonyHerman** [comment]
FWIW I'm using this already. Works great.

**Comment by jasonodonnell** [comment]
Hi @shahbazn, upon further review there's an error in your test code.  You are missing a closing brace on `server/standalone-StatefulSet: no extra containers added`.  This will need to be fixed prior to merging.

File: test/unit/server-statefulset.bats [hunk 1]
```diff
@@ -568,6 +568,86 @@ load _helpers
...
+  # Test that it defines it
+  local object=$(helm template \
+      -x templates/server-statefulset.yaml  \
+      . | tee /dev/stderr |
+      yq -r '.spec.template.spec.containers' | tee /dev/stderr)
+
+  local containers_count=$(echo $object |
+      yq -r 'length' | tee /dev/stderr)
+  [ "${containers_count}" = 1 ]
+
```
Comment (jasonodonnell) [inline test/unit/server-statefulset.bats]: Missing `}`

**Comment by iusergii** [comment]
This can be solution for https://github.com/hashicorp/vault-helm/issues/109.

**Pull request merged**